pkg/platform/tinkerbell/tinkerbell.go

```diff
@@ -288,3 +289,34 @@ func (c *Config) validateRequiredFields() hcl.Diagnostics {
 
 	return d
 }
+
+// CheckWorkerPoolLabelsAndTaints verifies that all worker pool labels
+// and taints are in correct format. Neither key nor value of labels
+// and taints map should have empty space in them.
+func (c *Config) CheckWorkerPoolLabelsAndTaints() hcl.Diagnostics {
```

Comment:
This function should not be exported so we do not pollute the exported API. Config struct already exports generic `Validate()`.